src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/AbstractTablesTab.java

```diff
@@ -385,7 +380,7 @@ private void configureVersionAlert() {
 					goToLatestSnapshot,
 					isLinkToCurrentSnapshotEnabled,
 					linkToCurrentSnapshotTooltipText);
-		} else if (entityBundle.getEntity() instanceof Dataset && !version.equals(latestSnapshotVersionNumber)) {
+		} else if (entityBundle.getEntity() instanceof Dataset && !((Dataset) entityBundle.getEntity()).getIsLatestVersion() && version != latestSnapshotVersionNumber) {
```

Comment:
This additional condition was always true, but this makes it more clear.

Reply:
@nickgros , could you clarify the cases (in a code comment) to clarify this logic?
I think the cases are:  latest/draft version, most recent snapshot, and older snapshot.
To confirm, is getIsLatestVersion() referring to the "draft" version, and latestSnapshotVersionNumber is the most recent snapshot?

Reply:
I commented + refactored the logic to try to clarify here: https://github.com/Sage-Bionetworks/SynapseWebClient/pull/4778/files/f6fc15f51af523c7bfc6a89d6020e37bb6f41887..0be3a2a7b0f04672354e792bd1d41913dfc1c7ac

The cases you listed are correct, and we also have to check if the Table is a Dataset